source/common/ssl/connection_impl.cc

```diff
@@ -214,6 +214,22 @@ Network::ConnectionImpl::IoResult ConnectionImpl::doWriteToSocket() {
 
 void ConnectionImpl::onConnected() { ASSERT(!handshake_complete_); }
 
+bool ConnectionImpl::isMtls() {
+  if (SSL_get_peer_certificate(ssl_.get())) {
```

Comment:
The implementation here is wrong in case of the connection is a client connection. We could call this `peerCertificatePresented` instead of isMtls, or check both certificate.

Also note the X509 pointer returned by `SSL_get_peer_certificate` need to be freed. Use `bssl::UniquePtr`.